app/proxy/handlers.go

```diff
@@ -40,18 +40,21 @@ func (rh *RequestHandler) Handle(w http.ResponseWriter, r *http.Request) {
 
 	var walletID string
 	if config.AccountsEnabled() {
-		retriever := users.NewWalletService()
-		auth := users.NewAuthenticator(retriever)
-		walletID, err = auth.GetWalletID(r)
+		q, err := NewQuery(body)
+		if err != nil || !methodInList(q.Method(), relaxedMethods) {
+			retriever := users.NewWalletService()
+			auth := users.NewAuthenticator(retriever)
+			walletID, err = auth.GetWalletID(r)
 
-		// TODO: Refactor error response creation
-		if err != nil {
-			response, _ := json.Marshal(NewErrorResponse(err.Error(), ErrAuthFailed))
-			w.Header().Set("Content-Type", "application/json; charset=utf-8")
-			w.WriteHeader(http.StatusOK)
-			w.Write(response)
-			monitor.CaptureRequestError(err, r, w)
-			return
+			// TODO: Refactor error response creation
+			if err != nil {
+				response, _ := json.Marshal(NewErrorResponse(err.Error(), ErrAuthFailed))
+				w.Header().Set("Content-Type", "application/json; charset=utf-8")
```

Comment:
maybe its time to refactor this https://github.com/lbryio/lbrytv/search?q=application%2Fjson&unscoped_q=application%2Fjson

Reply:
Agree, it's in the TODO as well.